### PR TITLE
Fix too many arguments error

### PR DESCRIPTION
**Pull request by Acentelles**
Fixes https://github.com/anoma/juvix/issues/1004

### CODE REVIEW

**Comment by andy-morris** [comment]
ah good catch, thanks!

**Comment by andy-morris** [comment]
though, maybe this and [the other place](https://github.com/anoma/juvix/blob/alberto/fix-too-many-arguments/library/Pipeline/src/Juvix/Pipeline/Compile.hs#L193-L195) could be factored out or something, since the source of the problem was two occurrences of the same logic?

**Review by andy-morris — APPROVED** [review]

**Comment by systemf-frank** [comment]
Huh both of the commits are done by drone? Something wrong with you git setup?

**Comment by cyberglot** [comment]
> Huh both of the commits are done by drone? Something wrong with you git setup?
When they said robots would take over programming I laughed, but they're indeed onto us.

File: library/Core/src/Juvix/Core/IR/CheckTerm.hs [hunk 1]
```diff
@@ -179,7 +180,11 @@ typeTerm' term ann@(Typed.Annotation σ ty) =
     Core.PrimTy t _ -> do
       requireZero σ
       void $ requirePrimStars (Param.primArity t) ty
-      let t' = App.Return {retTerm = t, retType = Param.PrimType $ Param.STAR :| []}
+      let t' =
+            App.Return
+              { retTerm = t,
+                retType = Param.PrimType $ NonEmpty.fromList $ replicate (fromIntegral (Param.primArity t + 1)) Param.STAR
```
Comment (systemf-frank) [inline library/Core/src/Juvix/Core/IR/CheckTerm.hs:186]: A recursive function of over `ty` should also work, I guess. Would that be nicer?

**Pull request merged**